queries/course_requests.js

```diff
@@ -1,31 +1,141 @@
+                    .json({
+                        number: Object.values(recordset.recordset[0])[0]
+                    });
+            }
+        );
+    },
+
+}
+
+/**
+ * Updates the status of the new course requests to pending (= 1).
+ * 
+ * @param {*} sql The mssql instance connected to the database currently used by the API.
+ */
+function updateCourseRequests(sql) {
+
     const request = new sql.Request();
 
-    request.query(
-      "SELECT COUNT(ID) FROM CourseRequests WHERE Status=0",
-      function (err, recordset) {
-        if (err) {
-          res.status(400).json({ error: err });
+    request
+        .query('update CourseRequests set status = 1 where status = 0', function (err) {
+            if (err) {
+                console.log(err);
+            } else {
+                console.log("New course request(s) have been updated.");
+            }
+        });
+}
+
+/**
+ * The method allows to get a well formatted result for the new course request route.
+ * Each course request record now has a property dateOptions that gathers all the date options in a list. 
+ * 
+ * @param {*} result This is the list of all the new course requests.
+ * @returns The arranged list of all the new course requests having a property date options.
+ */
+function formatResult(result) {
+
+    let listOfSubjects = [];
+    let resultWanted = [];
+
+    for (const newCourseRequest of result) {
+        if (!listOfSubjects.includes(newCourseRequest.Subject)) {
+            listOfSubjects.push(newCourseRequest.Subject);
         }
+    }
+
+    for (const subject of listOfSubjects) {
+        let informationToRetrieve = {
+            "ID": undefined,
+            "Subject": undefined,
+            "Frequency": undefined,
+            "LevelName": undefined,
+            "Money": undefined,
+            "Duration": undefined,
+            "DateOptionsID": undefined,
+            "DateOptions": undefined
+        };
+
+        let courseRequestsOfSameSubject = [];
+        for (courseRequest of result) {
+            if (courseRequest.Subject == subject) {
+                courseRequestsOfSameSubject.push(courseRequest);
+                informationToRetrieve.ID = courseRequest.ID;
+                informationToRetrieve.Subject = courseRequest.Subject;
+                informationToRetrieve.Frequency = courseRequest.Frequency;
+                informationToRetrieve.LevelName = courseRequest.LevelName;
+                informationToRetrieve.Money = courseRequest.Money;
+                informationToRetrieve.Duration = courseRequest.Duration;
+            }
+        }
+
+        let dateOptionsForCourseRequestOfSameSubject = [];
+        for (courseRequest of courseRequestsOfSameSubject) {
+            dateOptionsForCourseRequestOfSameSubject.push({ID: courseRequest.DateOptionsID, String: courseRequest.Day + " " + courseRequest.Time});
+        }
+
+        informationToRetrieve.DateOptions = dateOptionsForCourseRequestOfSameSubject;
```

Comment:
I don't see how that works but it may but I don't see why you do it.
You could do:
const processedCourseReqIDs = [];
for (const courseReq of courseRequests)
{
if (!courseReq.ID in processedCourseReqIDs)
const informationToRetrieve ={"ID": courseReq.ID, "Subject":courseReq.subject, ...., "dateOptions": [{"ID":DateOptions.ID, "String": XXXXX}]}
for (courseRequestWithDateOptions of courseRequests) {
if (courseRequestWithDateOptions.ID === courseReq.ID) {
informationToRetrieve.DateOptions.push({"ID":DateOptions.ID, "String":XXXXX});
}
}
processedCourseReqIDs.push(courseReq.ID);
}
}

I dunno if all of that exactly is proper JS but the logic is here. Bcs I don't understand why subjects should ever come into play

Reply:
@absirgue just added your logic of the data cleaning function to the code.